src/Transports/Azure/Wolverine.AzureServiceBus/Internal/AzureServiceBusSubscription.cs

```diff
@@ -125,4 +127,19 @@ public async ValueTask PurgeAsync(ILogger logger)
             { nameof(SubscriptionProperties.Status), props.Value.Status.ToString() },
         };
     }
+
+    public override async ValueTask InitializeAsync(ILogger logger)
+    {
+        if (_hasInitialized)
+        {
+            return;
+        }
+
+        if (Parent.AutoProvision)
+        {
+            await SetupAsync(logger);
+        }
+
```

Comment:
@jeremydmiller Not sure whether you want support for `AutoPurge` in here too?

e.g.
```
if (Parent.AutoPurgeAllQueues)
{
    await PurgeAsync(logger);
}
```

Reply:
I think we do. I'll add that real quick after I pull this in.